libsolidity/interface/StandardCompiler.cpp

```diff
@@ -887,10 +887,12 @@ Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSetting
 		string file = contractName.substr(0, colon);
 		string name = contractName.substr(colon + 1);
 
-		// ABI, documentation and metadata
+		// ABI, storage layout, documentation and metadata
 		Json::Value contractData(Json::objectValue);
 		if (isArtifactRequested(_inputsAndSettings.outputSelection, file, name, "abi", wildcardMatchesExperimental))
 			contractData["abi"] = compilerStack.contractABI(contractName);
+		if (isArtifactRequested(_inputsAndSettings.outputSelection, file, name, "storageLayout", wildcardMatchesExperimental))
```

Comment:
"wildcardMatchesExperimental" means that you only get storage layout if you request it explicitly, so you don't get it when requesting `"*"`. This is needed for yul ir and wasm because the compiler would otherwise throw an exception, but I think it is not needed here.

Reply:
Ah right